CodeStarterCamp_Week3/Type.swift

```diff
@@ -7,42 +7,81 @@
 
 import Foundation
 
-struct Person {
+class Person {
     var name: String
     var money: UInt
     
-    func buySomething() {
-        print("\(self.name) buy something!")
+    init(name: String, money: UInt) {
+        self.name = name
+        self.money = money
+    }
+    
+    func buyCoffee(at coffeeShop: CoffeeShop, coffee: Coffee) {
+        if self.money < coffee.price {
+            print("잔액이 \(coffee.price - self.money)원만큼 부족합니다.")
+        } else {
+            let isOrderCompleted = coffeeShop.order(coffee: coffee, nameOfCustomer: self.name)
+            self.money = isOrderCompleted ? (self.money - coffee.price) : self.money
+        }
+    }
+}
+
+
+class Barista: Person {
+    func makeCoffee(coffee: Coffee) -> Bool {
+        var finishedExtractCoffee = false
+        finishedExtractCoffee = true
+        
+        return finishedExtractCoffee
     }
 }
 
+
 class CoffeeShop {
-    var sales: UInt
-    var barista: Person?
+    var sales: UInt = 0
+    var barista: Barista?
     var menu: [Coffee]
-    let pickUpTable: String = "This is pick up table!"
-    
-    init() {
-        self.sales = 0
-        self.menu = [.americano(price: 4500), .cafeLatte(price: 5000)]
-    }
+    var pickUpTable: Coffee?
     
-    func takeOrder(item: Coffee?) {
-        if case .some(let orderedItem) = item {
-            print(orderedItem)
-        } else { print("주문을 받지 않았습니다.") }
+    init(menu: [Coffee]) {
+        self.menu = menu
     }
     
-    func makeCoffee() {
+    func order(coffee: Coffee, nameOfCustomer: String) -> Bool {
+        var isSucceed: Bool = false {
+            didSet {
+                print("\(nameOfCustomer) 님의 커피가 준비되었습니다. 픽업대에서 가져가주세요.")
+                pickUpTable = coffee
+            }
+        }
         
+        if let barista = self.barista {
+            isSucceed = barista.makeCoffee(coffee: coffee)
+        } else {
+            print("현재 카페에 바리스타가 없어서 커피를 만들 수 없습니다!")
+        }
+        
+        self.sales += coffee.price
+        return isSucceed
     }
 }
 
 
 enum Coffee {
-    case americano(price: UInt)
-    case cafeLatte(price: UInt)
-    case cappuccino(price: UInt)
-    case hotChocolate(price: UInt)
+    case americano
+    case cafeLatte
+    case cappuccino
+    case hotChocolate
+    
+    var price: UInt {
+        get {
+            switch self {
+            case .americano: return 4500
+            case .cafeLatte: return 5500
+            case .cappuccino: return 6000
+            case .hotChocolate: return 5000
+            }
+        }
+    }
```

Comment:
```suggestion
    var price: UInt {
          switch self {
          case .americano: return 4500
          case .cafeLatte: return 5500
          case .cappuccino: return 6000
          case .hotChocolate: return 5000
          }
    }
```

swift에서는 get만 존재할때는 get 키워드를 생략하는것을 권장합니다!